feat: Add build script for Gex

Ehsan-saradar opened the pull request.

jeronimoalbi commented on scripts/build-gex:
```diff
@@ -0,0 +1,40 @@
+set -e
+
+GIT_REPOSITORY="https://github.com/cosmos/gex.git"
+VERSION="v1.3.0"
```
Might be a good idea to read the version from command line variable in case we want to automate the process. Maybe `v1.3.0` could be the default when no command line version is available?

Ehsan-saradar replied:
At first i wanted to this but it seems that gex repository is not maintained regularly. They don't have semver versioning and also my last PR that got merged into the main branch is not versioned.

jeronimoalbi reviewed (approved):
Eventually would be a good improvement to use Gex as a Go dependency instead of a binary

Pantani commented:
Can you also create a GitHub pipeline to update the binaries eventually in case we have a new one? Like the nodetime

Ehsan-saradar commented:
> @Ehsan-saradar, Can you also create a GitHub pipeline to update the binaries eventually in case we have a new one? Like the nodetime

Good suggestion, i changed the script and added a workflow file like the one for nodetime. could you double check and see if everything is done correctly?

jeronimoalbi commented on gex/gex-amd64-darwin.tar.gz:
I think we can remove all the binaries because they will be rebuilt by the CI when this PR is merged into `main`

jeronimoalbi commented on .github/workflows/gen-gex-binaries.yml:
```diff
@@ -0,0 +1,44 @@
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        id: qemu
+        uses: docker/setup-qemu-action@v1
+        with:
+          image: tonistiigi/binfmt:latest
+          platforms: all
+
+      - uses: MOZGIII/install-ldid-action@v1
+        with:
+          tag: v2.1.5-procursus2
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '16'
```
Are these three necessary?

Ehsan-saradar replied:
Ah yes, My bad!
I'm assuming it should be replaced by `setup-go` action then?

Ehsan-saradar replied:
Done
